### PR TITLE
Display information about the speakers of a session

### PR DESCRIPTION
**Pull request by lukket**
## Checklist

-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have made corresponding changes to the documentation
-   [x] I have added tests that prove my fix is effective or that my feature works
-   [x] I have checked my code and corrected any misspellings
-   [x] My changes take different screen sizes into account
-   [x] My changes are backward compatible
    -   [x] I have not changed any parameter names
    -   [x] I have not changed any translation keys
-   [x] My changes conform to the contributing guidelines (`CONTRIBUTING.md`)

image

image

image

### CODE REVIEW

File: assets/styles/layouts/partials/speaker-avatar.css [hunk 1]
```diff
@@ -15,7 +15,6 @@
     opacity: 0.5;
     transition: border 0.3s ease;
     position: relative;
-    z-index: 1;
```
Comment (NataliiaPodp) [inline assets/styles/layouts/partials/speaker-avatar.css:18]: Could you please double-check if the hover works as expected?

File: layouts/sessions/single.html [hunk 2]
```diff
@@ -1,18 +1,67 @@
...
-            <h1 class="single-session-section__heading">
-                <div class="single-session-section__metadata">
-                    {{- if page.Params.track }}
-                        <span class="single-session-section__hashtag">
-                            {{- partial "elements/hashtag.html" (dict "hashtag" page.Params.track) }}
-                        </span>
-                    {{- end }}
-                    {{- partial "elements/date-and-time.html" (dict "date" page.Params.startsAt ) }}
+        <section class="section ">
+            <div class="single-session-section-container">
```
Comment (NataliiaPodp) [inline layouts/sessions/single.html:4]: "container" is already an element, shouldn't it be "single-session-section__container"

File: layouts/sessions/single.html [hunk 3]
```diff
@@ -1,18 +1,67 @@
...
+                <div class="single-session-section-spacer"></div>
```
Comment (NataliiaPodp) [inline layouts/sessions/single.html:5]: "single-session-section__spacer" ?

File: tests/featured-speaker.spec.ts [hunk 4]
```diff
@@ -1,21 +1,35 @@
...
+    test(`Should provide link to page of featured speaker ${fullName} on home page`, async ({ page }) => {
         await page.goto('/');
-        await page.getByRole('menu', { name: 'Featured Speakers' }).getByRole('link', { name: name }).click();
-        await expect(page.getByRole('heading', { name: name })).toBeVisible();
+        const featuredSpeakersSection = page.getByRole('region', { name: 'Featured Speakers' });
+        const featuredSpeakersMenu = featuredSpeakersSection.getByRole('menu', { name: 'Featured Speakers' });
+        const featuredSpeakersMenuItem = featuredSpeakersMenu.getByRole('menuitem', { name: fullName });
+        await expect(featuredSpeakersMenuItem.getByText(fullName)).toBeVisible();
+        await expect(featuredSpeakersMenuItem.getByText(tagLine)).toBeVisible();
+        await featuredSpeakersMenuItem.getByRole('link', { name: fullName }).click();
```
Comment (NataliiaPodp) [inline tests/featured-speaker.spec.ts:32]: I would say this could already be a separate test. We don't test here whether the link was provided; we check if the link is working, no?

File: assets/styles/layouts/partials/speaker-avatar.css (hunk 1 above)
Comment (lukket) [reply assets/styles/layouts/partials/speaker-avatar.css:18]: Sure (pointer not visible on screenshot):

image

File: layouts/sessions/single.html (hunk 3 above)
Comment (lukket) [reply layouts/sessions/single.html]: changed

File: layouts/sessions/single.html (hunk 2 above)
Comment (lukket) [reply layouts/sessions/single.html]: changed

File: tests/featured-speaker.spec.ts (hunk 4 above)
Comment (lukket) [reply tests/featured-speaker.spec.ts:32]: I think it depends. Other than with unit or component tests, my understanding of the best practices of e2e tests is to test whole features of the website. More precise, multiple assertions are totally fine, because tools like Playwright or Cypress can give very precise information about failed assertions. And more tests slow down the tests, whereas multiple assertions in on test are fast.  In this case, the feature is to provide a link to every featured speaker with some information. And, of course, the link should also work. 

TLDR: From my understanding, the check if the link is working should not be separated. 😅

---

However, I check here if the full bio is available on the speaker's page. That's not really part of this feature. In other tests, I check just the URL.

**Review by NataliiaPodp — APPROVED** [review]
